### PR TITLE
Modifies sparks to use spark_at.

### PR DESCRIPTION
**Pull request by MarinaGryphon**
Replaces direct usage of spark systems to spark_at.

### CODE REVIEW

File: code/game/objects/items/devices/chameleonproj.dm [hunk 1]
```diff
@@ -65,10 +65,7 @@
 
 /obj/item/chameleon/proc/disrupt(var/delete_dummy = 1)
 	if(active_dummy)
-		var/datum/effect/effect/system/spark_spread/spark_system = new /datum/effect/effect/system/spark_spread
-		spark_system.set_up(5, 0, src)
-		spark_system.attach(src)
-		spark_system.start()
+		spark_at(src, amount = 2, cardinal_only = TRUE, holder = src)
```
Comment (MistakeNot4892) [inline code/game/objects/items/devices/chameleonproj.dm]: Shouldn't this be `amount = 5`?

File: code/modules/hydroponics/seed.dm [hunk 2]
```diff
@@ -361,9 +361,7 @@
 
 		var/turf/T = get_random_turf_in_range(target, outer_teleport_radius, inner_teleport_radius)
 		if(T)
-			var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
-			s.set_up(3, 1, get_turf(target))
-			s.start()
+			spark_at(target, cardinal_only = TRUE)
```
Comment (MistakeNot4892) [inline code/modules/hydroponics/seed.dm:364]: Missing `amount`?

File: code/modules/mechs/mech_damage.dm [hunk 3]
```diff
@@ -99,8 +99,7 @@
 			radiation += damage
 
 	if((damagetype == BRUTE || damagetype == BURN) && prob(25+(damage*2)))
-		sparks.set_up(3,0,src)
-		sparks.start()
+		spark_at(src)
```
Comment (MistakeNot4892) [inline code/modules/mechs/mech_damage.dm:102]: No `amount`?

File: code/modules/modular_computers/ntos/components.dm [hunk 4]
```diff
@@ -63,8 +63,8 @@
 /datum/extension/interactive/ntos/proc/voltage_overload()
 	var/atom/A = holder
 	if(istype(A))
-		var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
-		s.set_up(10, 1, A.loc)
+		var/datum/effect/effect/system/spark_spread/s = new
+		s.set_up(10, 1, A)
```
Comment (MistakeNot4892) [inline code/modules/modular_computers/ntos/components.dm]: Not using `spark_at` here?

File: code/modules/hydroponics/seed.dm (hunk 2 above)
Comment (MarinaGryphon) [reply code/modules/hydroponics/seed.dm:364]: Default is 3.

File: code/modules/mechs/mech_damage.dm (hunk 3 above)
Comment (MarinaGryphon) [reply code/modules/mechs/mech_damage.dm:102]: Again, default is 3, but I can make it explicit if you'd like.

Comment (MistakeNot4892) [reply code/modules/mechs/mech_damage.dm:102]: Discussed on Discord, needs to either use amount=3 or the other amount=3 should be removed.

File: code/modules/hydroponics/seed.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/modules/hydroponics/seed.dm:364]: See below comment.

File: code/modules/modular_computers/ntos/components.dm (hunk 4 above)
Comment (MistakeNot4892) [reply code/modules/modular_computers/ntos/components.dm]: Should delete this and the above line now that you're using spark_at().

File: code/game/objects/items/devices/chameleonproj.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/game/objects/items/devices/chameleonproj.dm]: Jade pls it's one character.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**